### PR TITLE
hv: correct the way to check if a MSR is a fixed MTRR register

### PR DESCRIPTION
**Pull request by ZideChen0**
The fixed MTRR MSR numbers are not contiguous, so it's not correct
to justify it by checking if it falls in certain range.

This patch fixes this issue by removing is_fixed_range_mtrr()
and use get_index_of_fixed_mtrr() to loop fixed_mtrr_map[]
and compare individual MSR values.

Also removed the unused function get_subrange_end_of_fixed_mtrr()

Signed-off-by: Zide Chen <zide.chen@intel.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: hypervisor/arch/x86/mtrr.c [hunk 1]
```diff
@@ -37,21 +37,17 @@ static struct fixed_range_mtrr_maps fixed_mtrr_map[FIXED_RANGE_MTRR_NUM] = {
...
 
-static bool is_fixed_range_mtrr(uint32_t msr)
-{
-	return (msr >= fixed_mtrr_map[0].msr)
-		&& (msr <= fixed_mtrr_map[FIXED_RANGE_MTRR_NUM - 1U].msr);
-}
-
-static uint32_t get_index_of_fixed_mtrr(uint32_t msr)
+static bool get_index_of_fixed_mtrr(uint32_t msr, int *index)
 {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/mtrr.c]: I prefer u return index here, but define an XXX_INVALID_INDEX...  The caller need to check if it is INVALID_INDEX

Comment (ZideChen0) [reply hypervisor/arch/x86/mtrr.c]: Sure, will do

**Comment by lijinxia** [comment]
Verified+1

**Pull request merged**